scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -1270,7 +1270,8 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
   }
 
   def makeTupleType(body: List[Type]): Type = {
-    def emptyTupleResult = if (dialect.allowLiteralUnitType) Lit.Unit() else unreachable
+    def emptyTupleResult = if (dialect.allowLiteralUnitType) Lit.Unit()
```

Comment:
if no longer supported I would remove if and just inline syntaxError

Reply:
Good idea, thanks!